### ISSUE TITLE
Ambiguous glob imports have functions shadow consts.

### ISSUE THREAD

**Issue by theemathas** [rust-lang/rust#149416]
I tried this code:

```rust
mod a {
    pub fn foo() {
        println!("a");
    }
}
mod b {
    const foo: i32 = 1;
}

use a::*;
use b::*;

fn main() {
    let f = foo;
    f();
}
```

I expected a compile error, since the glob import is ambiguous. Instead, the code compiles, and prints `a`.

### Meta

Reproducible on the playground with version `1.93.0-nightly (2025-11-27 c86564c412a5949088a5)`

**Comment by petrochenkov** [comment]
`const foo` here is private, so it gets filtered away by the glob import.
If you make it public, the ambiguity error will appear.
